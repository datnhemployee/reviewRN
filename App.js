import React, {Component} from 'react';
// mobx
import {Provider} from 'mobx-react';
// screens
import Home from './src/screens/home';
// stores
import userStore from './src/stores/UserStore';

const stores = {
  userStore,
};

class App extends Component {
  constructor(props) {
    super(props);
  }

  render() {
    return (
      <Provider {...stores}>
        <Home />
      </Provider>
    );
  }
}

export default App;
