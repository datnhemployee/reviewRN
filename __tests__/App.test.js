import 'react-native';
import React from 'react';
import App from '../App';
/**
 * @tutorial https://www.youtube.com/watch?v=lgkyhEHIC_c
 */
import renderer from 'react-test-renderer';

describe('Home.js/', () => {
  it('render', () => {
    // Set up(fetch data, connect DB, create instances etc)
    let tree = renderer.create(<App />);
    // Run (create JSON layout, run function)
    tree = tree.toJSON();
    // Asset (expected result)
    expect(tree).toMatchSnapshot();
  });
});
