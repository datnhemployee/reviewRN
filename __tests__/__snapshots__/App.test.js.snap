// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home.js/ render 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "flex": 1,
        "opacity": 1,
      }
    }
  >
    <Text>
      Press Me
    </Text>
  </View>
</RCTSafeAreaView>
`;
